nav2_controller/plugins/feasible_path_handler.cpp

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2022 Samsung Research America, @artofnothingness Alexey Budyakov
+// Copyright (c) 2023 Dexory
+// Copyright (c) 2023 Open Navigation LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include "angles/angles.h"
+#include "pluginlib/class_list_macros.hpp"
+#include "nav2_controller/plugins/feasible_path_handler.hpp"
+#include "nav2_util/path_utils.hpp"
+#include "nav2_util/geometry_utils.hpp"
+#include "nav2_core/controller_exceptions.hpp"
+
+
+using rcl_interfaces::msg::ParameterType;
+using std::placeholders::_1;
+
+namespace nav2_controller
+{
+using nav2_util::geometry_utils::euclidean_distance;
+
+void FeasiblePathHandler::initialize(
+  const nav2::LifecycleNode::WeakPtr & parent,
+  const rclcpp::Logger & logger,
+  const std::string & plugin_name,
+  const std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros,
+  std::shared_ptr<tf2_ros::Buffer> tf)
+{
+  logger_ = logger;
+  plugin_name_ = plugin_name;
+  auto node = parent.lock();
+  costmap_ros_ = costmap_ros;
+  tf_ = tf;
+  transform_tolerance_ = costmap_ros_->getTransformTolerance();
+  reject_unit_path_ = node->declare_or_get_parameter(plugin_name +
+    ".reject_unit_path", false);
+  max_robot_pose_search_dist_ = node->declare_or_get_parameter(plugin_name +
+    ".max_robot_pose_search_dist", getCostmapMaxExtent());
+  prune_distance_ = node->declare_or_get_parameter(plugin_name + ".prune_distance",
+    2.0);
+  enforce_path_inversion_ = node->declare_or_get_parameter(plugin_name +
+    ".enforce_path_inversion", false);
+  enforce_path_rotation_ = node->declare_or_get_parameter(plugin_name +
+    ".enforce_path_rotation", false);
+  inversion_xy_tolerance_ = node->declare_or_get_parameter(plugin_name + ".inversion_xy_tolerance",
```

Comment:
I dont know what's going on with our linting right now, but this would break it. Anytime there's a first `(` and a line break, it'll want you to move the contents of that scope to a new line, i.e. 

```
  inversion_xy_tolerance_ = node->declare_or_get_parameter(
    plugin_name + ".inversion_xy_tolerance", 0.2);
```

Fix that here and in other places now so that once that's working properly we have less errors to deal with. Something is definitely going on with the ament linters, both our CircleCI and the independently run GitHub Actions linter jobs are asleep.